### ISSUE TITLE
Is there an sample using Core Data?

### ISSUE THREAD

**Issue by RodneyJ725** [kirbyt/KTPhotoBrowser#37]
Hello, I'm new, as my question may show, but how does one use the KTPhotoBrowser using Core Data? I see the local example and web examples (web and flickr).  Is there a sample using Core Data?  Sorry to ask here, but I did not see any other place to leave questions or simple comments.

**Comment by kirbyt** [comment]
No, there isn't a Core Data sample. I'll add it to my to-do list, but it will be some time before I can get to it.

Just so you know, KTPhotoBrowser doesn't actually know anything about the photos. Instead, your app provides a data source to KTPhotoBrowser. This allows the application to use any implementation it desires for storing, retrieving, etc photos.

A word of warning about using Core Data, depending on the size of the image files, you may want to store the actual image files (.png, .jpeg) outside of Core Data.  See http://stackoverflow.com/questions/2573072/coredata-store-images-to-db-or-not for more. The good news is with iOS 5, Core Data now has the external storage flag (see here: http://stackoverflow.com/questions/7924840/storing-uiimage-in-core-data-with-the-new-external-storage-flag). This allows Core Data to decide where the image file is stored. But it's not without it's own problems (see here: http://stackoverflow.com/questions/7930427/error-uiimage-deleteexternalreferencefrompermanentlocation-unrecognized-se).

-KIRBY

**Comment by Pilsner6910** [comment]
i think it would integrate nicely with CoreData, now that i have the external storage working. seems there were some bugs in it. So i think this KTPhotoBrowser would work great for me and that parts of it i want to add to my project. I've researched others like Three20 and its so bloated for the small portion i actually want to use in my project. then i found KTPhotoBrowser...so happy i did, i haven't actually included in my project yet but it sounds like it will work great.

**Comment by langk** [comment]
I only can agree with kirbyt - I am using CoreData and had several performance issues when storing image data in MOC. Didn't try the external storage flag though - so if somebody has used this I would be happy to hear about it.

**Comment by Pilsner6910** [comment]
well i don't have Allows Ext. Storage set up with KTPhotoBrowser yet but i do have Allows Ext. Storage working in my project and it works great. my tableView with thumbnails in each row are smooth as butter.
I did find there was still bugs in this new option, Im working with Lion of course, it didn't work in Snow Leopard for me.

I do have the portion of KTPhotoBrowser in my project working so far and Im using ARC, a bit of fixing but its working nicely. Im actually surprised it was so smooth to add. i just joined the git today so i haven't got my email set up yet but I can show you what i have when i do. i don't think it will be that hard to get Allows External Storage to integrate with KT.

Im loving KT so far because everything else like Three20 was much more then i needed and didn't want to add to my project.
